File: tests/fixtures/csharp/Service.cs
using System;
using System.Collections.Generic;

// User record
public record User(int Id, string Name, string Email);

// Repository interface
public interface IRepository
{
    User? FindById(int id);
    void Save(User user);
    int Count();
}

// Logger interface
public interface ILogger
{
    void Log(string message);
}

// UserService class
public class UserService
{
    private readonly IRepository _repo;
    private readonly ILogger _logger;

    public UserService(IRepository repo, ILogger logger)
    {
        _repo = repo;
        _logger = logger;
    }

    public User? FindById(int id)
    {
        _logger.Log($"Finding user {id}");
        return _repo.FindById(id);
    }

    public void Save(User user) => _repo.Save(user);
    public int Count() => _repo.Count();
}

// AbstractController
public abstract class AbstractController
{
    protected string ViewPath { get; }

    protected AbstractController(string viewPath)
    {
        ViewPath = viewPath;
    }

    public string Render(string template) => $"{ViewPath}/{template}";
}

// UserController extends AbstractController
public class UserController : AbstractController
{
    private readonly UserService _userService;

    public UserController(UserService userService)
        : base("users")
    {
        _userService = userService;
    }

    public string Index() => Render("index");
    public string Show(int id)
    {
        var user = _userService.FindById(id);
        return Render("show");
    }
}

// Utils namespace
namespace Utils
{
    public static class StringHelper
    {
        public static string FormatName(string name) => $"User: {name}";
    }
}

File: tests/fixtures/cpp/service.cpp
#include <string>
#include <vector>
#include <iostream>

// User class
class User {
public:
    int id;
    std::string name;
    std::string email;

    User(int id, const std::string& name, const std::string& email)
        : id(id), name(name), email(email) {}

    int getId() const { return id; }
    std::string getName() const { return name; }
};

// Repository abstract class
class Repository {
public:
    virtual ~Repository() = default;
    virtual User* findById(int id) = 0;
    virtual void save(User* user) = 0;
    virtual int count() = 0;
};

// Logger class
class Logger {
private:
    std::string prefix;
public:
    Logger(const std::string& p) : prefix(p) {}
    void log(const std::string& message) {
        std::cout << "[" << prefix << "] " << message << std::endl;
    }
};

// UserService class
class UserService {
private:
    Repository* repo;
    Logger* logger;
public:
    UserService(Repository* repo, Logger* logger)
        : repo(repo), logger(logger) {}

    User* findById(int id) {
        logger->log("Finding user " + std::to_string(id));
        return repo->findById(id);
    }

    void save(User* user) {
        repo->save(user);
    }

    int count() {
        return repo->count();
    }
};

// AbstractController
class AbstractController {
protected:
    std::string viewPath;
public:
    AbstractController(const std::string& path) : viewPath(path) {}
    virtual ~AbstractController() = default;
    std::string render(const std::string& templateName) {
        return viewPath + "/" + templateName;
    }
};

// UserController extends AbstractController
class UserController : public AbstractController {
private:
    UserService* userService;
public:
    UserController(UserService* service)
        : AbstractController("users"), userService(service) {}

    std::string index() { return render("index"); }
    std::string show(int id) {
        User* user = userService->findById(id);
        return render("show");
    }
};

// Utils namespace
namespace Utils {
    std::string formatName(const std::string& name) {
        return "User: " + name;
    }

    class StringHelper {
    public:
        static std::string trim(const std::string& s) { return s; }
    };
}
